rootless/specconv/specconv_linux.go

```diff
@@ -70,5 +74,63 @@ func toRootless(spec *specs.Spec, v2Controllers []string, currentOOMScoreAdj int
 	if spec.Process.OOMScoreAdj != nil && *spec.Process.OOMScoreAdj < currentOOMScoreAdj {
 		*spec.Process.OOMScoreAdj = currentOOMScoreAdj
 	}
+
+	// Fix up /proc if --pid=host
+	if ok, err := isPidHost(spec); err != nil {
+		return err
+	} else if ok {
+		if err := bindMountHostProcfs(spec); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isPidHost(spec *specs.Spec) (bool, error) {
+	for _, ns := range spec.Linux.Namespaces {
+		if ns.Type == specs.PIDNamespace {
+			if ns.Path == "" {
+				return false, nil
+			}
+			pidNS, err := os.Readlink(ns.Path)
+			if err != nil {
+				return false, err
+			}
+			selfPidNS, err := os.Readlink("/proc/self/ns/pid")
+			if err != nil {
+				return false, err
+			}
+			return pidNS == selfPidNS, nil
+		}
+	}
+	return true, nil
+}
+
+func bindMountHostProcfs(spec *specs.Spec) error {
+	// Replace procfs mount with rbind
+	// https://github.com/containers/podman/blob/v3.0.0-rc1/pkg/specgen/generate/oci.go#L248-L257
+	for i, m := range spec.Mounts {
+		if filepath.Clean(m.Destination) == "/proc" {
```

Comment:
This can use `path.Clean` as well (instead of `filepath`. Do we want a fast-path here in case no cleaning is needed?

```suggestion
		if m.Destination == "/proc" || path.Clean(m.Destination) == "/proc" {
```

Reply:
That makes the code longer and I feel less readable, but just my feeling

Reply:
Too bad we don't have a spec requirement that the paths should be absolute and cleaned.

Hmm, a somewhat similar code in buildkit do not have `path.Clean()`:

https://github.com/moby/buildkit/blob/master/util/rootless/specconv/specconv_linux.go#L28-L29

So I'm not sure whether it is needed here.